eth-contracts/contracts/service/storage/ServiceProviderStorage.sol

```diff
@@ -147,6 +147,32 @@ contract ServiceProviderStorage is RegistryContract {
         serviceProviderInfo[_serviceType][spID].delegateOwnerWallet = _updatedDelegateOwnerWallet;
     }
 
+    function updateEndpoint(
+        address _owner,
+        bytes32 _serviceType,
+        string calldata _oldEndpoint,
+        string calldata _newEndpoint
+    ) external onlyRegistrant(CALLER_REGISTRY_KEY) returns (uint spID)
+    {
+        uint spId = this.getServiceProviderIdFromEndpoint(_oldEndpoint);
+
+        require(
+            serviceProviderInfo[_serviceType][spId].owner == _owner,
+            "Invalid update endpoint operation, wrong owner");
+
+        ServiceProvider memory sp = serviceProviderInfo[_serviceType][spId];
```

Comment:
i think this should validate that the old endpoint equals whats in storage

Reply:
and not update if they don't match

Reply:
the reason this matters now is bc you can have multiple endpoints per owner

Reply:
if i understand you correctly have a require here with something like this?

```
require(
sp.endpoint == _oldEndpoint,
"Old endpoint doesn't match what's registered for the service provider"
);
```

Reply:
yes

Reply:
good catch, done